lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -115,18 +116,37 @@ public void writeBlock(ColumnChunk columnChunk)
         // write values
         primitiveValueWriter.write(columnChunk.getBlock());
 
-        // write definition levels
-        Iterator<Integer> defIterator = DefLevelIterables.getIterator(ImmutableList.<DefLevelIterable>builder()
-                .addAll(columnChunk.getDefLevelIterables())
-                .add(DefLevelIterables.of(columnChunk.getBlock(), maxDefinitionLevel))
-                .build());
-        while (defIterator.hasNext()) {
-            int next = defIterator.next();
-            definitionLevelWriter.writeInteger(next);
-            if (next != maxDefinitionLevel) {
-                currentPageNullCounts++;
+        if (columnChunk.getDefLevelIterables().isEmpty()) {
+            // write definition levels for flat data types
+            Block block = columnChunk.getBlock();
+            if (!block.mayHaveNull()) {
+                for (int position = 0; position < block.getPositionCount(); position++) {
```

Comment:
Can't we skip writing definition levels if maxDef is 0? Or maybe some readers will not support that?

Reply:
We could have skipped only if the type was a REQUIRED flat type, but AFAIK we don't create that scenario. For an OPTIONAL flat type, the reader would be right to expect definition levels.

Reply:
We may eventually want to add better support for Delta Lake's constraints, which allow for specifying non-null columns. But we don't have that yet

Reply:
With max definition level 0, the ValuesWriter will be a DevNullValuesWriter which does nothing internally, so we probably don't need to optimise that case explicitly.
Also max definition level will be 0 only when the type is flat and required (which we don't do today but might in the future).